plugins/cilium-cni/lib/deletion_queue.go

```diff
@@ -34,22 +52,43 @@ const timeoutSeconds = 10
 // the maximum number of queued deletions allowed, to protect against kubelet insanity
 const maxDeletionFiles = 256
 
+var (
+	errServiceUnavailable = errors.New("cilium api not available")
+)
+
 // NewDeletionFallbackClient creates a client that will either issue an EndpointDelete
 // request via the api, *or* queue one in a temporary directory.
-// To prevent race conditions, the logic is:
-// 1. Try and connect to the socket. if that succeeds, done
-// 2. Otherwise, take a shared lock on the delete queue directory
-// 3. Once we get the lock, check to see if the socket now exists
-// 4. If it exists, drop the lock and use the api
 func NewDeletionFallbackClient(logger *slog.Logger) (*DeletionFallbackClient, error) {
 	dc := &DeletionFallbackClient{
 		logger: logger,
+
+		deleteQueueDir:      defaults.DeleteQueueDir,
+		deleteQueueLockfile: defaults.DeleteQueueLockfile,
+
+		newCiliumClientFn: newCiliumClient,
 	}
 
+	err := dc.initialize()
```

Comment:
nit: at this point, we can just move `dc.initialize()` to `EndpointDeleteMany`. It will simplify testing as well.

Reply:
I have updated the PR to adapt to this review. There is one custom handling we have to add to `EndpointDeleteMany` call in cni delete processing to make sure that if the Delete was not processed by cilium(connect failed + delete lock/delete queue failed), we propagate the error to kubelet to retry. This matches the behavior we had before.
The flow now looks more clear to me, let me know what you think.